### DISCUSSION TITLE
Jakarta EdgeX 2.1 - is it support MQTT 5 or not?

### DISCUSSION THREAD

**Discussion by Abhijithm2447** [edgexfoundry/community#159] [General]
Jakarta EdgeX 2.1 - is it support MQTT 5 or not?

**Comment by bnevis-i** [comment]
No.  The MQTT client library EdgeX is using only supports MQTT 3.1, according to its documentation.

https://github.com/edgexfoundry/go-mod-messaging/blob/v2.1.0/go.mod

```
module github.com/edgexfoundry/go-mod-messaging/v2

require (
	github.com/eclipse/paho.mqtt.golang v1.3.5
	github.com/go-redis/redis/v7 v7.3.0
	github.com/pebbe/zmq4 v1.2.7
	github.com/stretchr/testify v1.7.0
)

go 1.16
```

https://github.com/eclipse/paho.mqtt.golang

> This repository contains the source code for the [Eclipse Paho](https://eclipse.org/paho) MQTT 3.1/3.11 Go client library.
> This code builds a library which enable applications to connect to an [MQTT](https://mqtt.org/) broker to publish messages, and to subscribe to topics and receive published messages.
> This library supports a fully asynchronous mode of operation.
> A client supporting MQTT V5 is [also available](https://github.com/eclipse/paho.golang).

**Comment by lenny-goodell** [comment]
I have created an issue to add support for MQTT5

**Comment by Abhijithm2447** [comment]
Could you provide insight into the support for shared subscriptions in EdgexFoundry's Jakarta 2.1 release, considering that MQTT 5's "shared subscription" feature, as indicated in this documentation (https://docs.hivemq.com/hivemq/4.17/user-guide/shared-subscriptions.html), appears exclusive to MQTT 5? I've observed that even though EdgexFoundry Jakarta 2.1 doesn't support MQTT 5, I've managed to achieve shared subscriptions using the "hive mq" broker and the topic "\$share/group/topic". This situation has left me somewhat perplexed. Could someone elucidate the rationale behind this capability in the context of my observations?

**Comment by lenny-goodell** [comment]
The Hive documentation at the link above clearly states that it supports shared subscriptions for _**MQTT 3** and MQTT 5 clients_

```
Shared subscriptions allow the MQTT broker to balance the load of a single subscription across multiple MQTT clients. HiveMQ supports shared subscriptions for MQTT 3 and MQTT 5 clients. 
```
